packages/lesswrong/server/manualMigrations/2024-06-05-rewriteOldReviewBotComments.ts

```diff
@@ -0,0 +1,51 @@
+import { Comments } from "@/lib/collections/comments";
+import { getSqlClientOrThrow } from "../sql/sqlClient";
+import { createAdminContext, updateMutator } from "../vulcan-lib";
+import { registerMigration } from "./migrationUtils";
+
+registerMigration({
+  name: "rewriteOldReviewBotComments",
+  dateWritten: "2024-06-05",
+  idempotent: true,
+  action: async () => {
+    const sql = getSqlClientOrThrow()
+
+    const context = createAdminContext()
+
+    const oldReviewBotComments: DbComment[] = await sql.many(`
+      SELECT * FROM "Comments"
+      WHERE "userId" = 'tBchiz3RM7rPwujrJ' AND "deleted" = FALSE
+    `);
+
+    const reviewBotUser: DbUser = await sql.one(`
+      SELECT * FROM "Users"
+      WHERE "_id" = 'tBchiz3RM7rPwujrJ'
+    `);
+
+    for (const comment of oldReviewBotComments) {
+      if (!comment.contents?.html) throw new Error("oh no!", String(comment.contents));
+      const manifoldUrl = comment.contents.html.match(/https:\/\/manifold.markets\/LessWrong\/([A-Za-z\d-])+/)?.[0];
+      const year = comment.contents.html.match(`but will be at the end of (\\d+)`)?.[1];
+      if (!manifoldUrl || !year) continue
```

Comment:
This case probably warrants a console log, since it means either the regex failed or there's a review bot comment which doesn't have the thing we expected in it